### DISCUSSION TITLE
Expecting property name enclosed in double quotes

### DISCUSSION THREAD

**Discussion by madtatu-development** [msiemens/tinydb#435] [Q&A]
Hi, 

May I ask your help ?

I am using Tinydb in python in a super simple way:

`#----------------------------------- write data to db
def to_db (name, value):

    db = TinyDB ("data.dat")        
    db.upsert ({"name": name, "value": value}, Query ().name == name)
#-----------------------------------`

`#----------------------------------- read daat from db
def from_db (name):

    db = TinyDB ("data.dat")
    exist = db.search (Query ()["name"] == name)  
    
    if exist == []:   

        return 0

    else :
        
        value = db.get (Query ()["name"] == name)

        return value.get ("value")    
#-----------------------------------`

where:

- name parameter is always a string with lowercase alphabetic letters and underscore, e.g. : item_name
- value parameter is always a string with numerical number, with either dot or double dot, e.g.: 13.45 or 13:45
- saved file shows perfect json structure. 

**but strings NEVER contain quote or double quote**

Still is random and sudden occasions, I ran into the following exception :

**"Expecting property name enclosed in double quotes"**

OR

**"json.decoder.JSONDecodeError: Expecting ',' delimiter: line 1 column 649 (char 648)"**

at the line of:

`db.upsert ({"name": name, "value": value}, Query ().name == name)`

I read all related StackOwerflow articles, but I feel they dont refer to my problem, as I properly use double quote with "name" and "value", and the parameters never contains single or double quotes.

Could you help me please what am I overlooking here ?

Many thanks for your time and help !

**Comment by msiemens** [comment]
Is it possible some other process writes to the database file? The exception message would hint to me that somehow the `data.dat` file contains a Python dict, not a JSON object.

Could you post an example of your database file at the time when the exception occurs?

**Comment by vvvhung** [comment]
I have the same problem, and the detail tracing:
`
File "/usr/local/lib/python3.7/site-packages/tinydb/table.py", line 570, in __len__
    tables = self._storage.read()
  File "/usr/local/lib/python3.7/site-packages/tinydb/storages.py", line 125, in read
    return json.load(self._handle)
  File "/usr/local/lib/python3.7/json/__init__.py", line 296, in load
    parse_constant=parse_constant, object_pairs_hook=object_pairs_hook, **kw)
  File "/usr/local/lib/python3.7/json/__init__.py", line 348, in loads
    return _default_decoder.decode(s)
  File "/usr/local/lib/python3.7/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/local/lib/python3.7/json/decoder.py", line 353, in raw_decode
    obj, end = self.scan_once(s, idx)
json.decoder.JSONDecodeError: Expecting property name enclosed in double quotes: line 273 column 6 (char 11957)
`



The problem happen sometime.

**Reply by msiemens** [reply]
Do you still have the JSON file that caused the issue? Without the file it's difficult to pin this down
